internal/exec/stages/files/filesystemEntries.go

```diff
@@ -186,12 +189,24 @@ func (s stage) getOrderedCreationList(config types.Config) []filesystemEntry {
 		entries = append(entries, fileEntry(f))
 	}
 
-	for _, sy := range config.Storage.Links {
-		entries = append(entries, linkEntry(sy))
+	for _, l := range config.Storage.Links {
+		entries = append(entries, linkEntry(l))
 	}
-	sort.Slice(entries, func(i, j int) bool { return util.Depth(entries[i].getPath()) < util.Depth(entries[j].getPath()) })
+	sort.Slice(entries, func(i, j int) bool {
+		ipath, err2 := s.JoinPath(entries[i].getPath())
+		if err2 != nil && err == nil {
+			err = err2
```

Comment:
Is there a reason that both of these cases don't immediately return?

Reply:
they're in a closure.

Reply:
Yes, but in the failure case it doesn't seem like it matters if you return an immediate `true` or `false` and it will also prevent `util.Depth` from receiving a blank string (which it doesn't seem to like as it doesn't handle the case where `path.Clean` returns `.` when it receives an empty string).

Reply:
Returning early isn't super useful here. That function will be called over and over. It'd be bogus results either way since either we return a static true/false that messes with ordering or we return the bogus result of comparing something via bad calls to util.Depth.

We could panic and recover but that's also not great.

We could resolve everything now changing the paths in the dir/link/file structs and not when running, but that's a much bigger refactor.